cirq/experiments/qubit_characterizations.py

```diff
@@ -0,0 +1,639 @@
+import itertools
+from typing import Sequence, Tuple, Iterator
+from cirq import circuits, devices, ops, protocols, sim, study, value
+import numpy as np
+from matplotlib import pyplot
+from mpl_toolkits.mplot3d import Axes3D
+
+
+class RabiResult:
+    """Results from a Rabi oscillation experiment."""
+
+    def __init__(self, rabi_angles: Sequence[float],
+                 excited_state_probabilities: Sequence[float]):
+        """
+        Args:
+            rabi_angles: The rotation angles of the qubit around the x-axis
+                of the Bloch sphere.
+            excited_state_probabilities: The corresponding probabilities that
+                the qubit is in the excited state.
+        """
+        self._rabi_angles = rabi_angles
+        self._excited_state_probs = excited_state_probabilities
+
+    @property
+    def data(self) -> Sequence[Tuple[float, float]]:
+        """Returns a sequence of tuple pairs with the first item being a Rabi
+        angle and the second item being the corresponding excited state
+        probability.
+        """
+        return [(angle, prob) for angle, prob in zip(self._rabi_angles,
+                                                     self._excited_state_probs)]
+
+    def plot(self) -> None:
+        """Plots excited state probability vs the Rabi angle (angle of rotation
+        around the x-axis).
+        """
+        fig = pyplot.figure()
+        pyplot.plot(self._rabi_angles, self._excited_state_probs, 'ro-',
+                    figure=fig)
+        pyplot.xlabel(r"Rabi Angle ($\pi$)")
+        pyplot.ylabel('Excited State Probability')
+
+
+class RBResult:
+    """Results from a randomized benchmarking experiment."""
+
+    def __init__(self, numbers_cliffords: Sequence[int],
+                 ground_state_probabilities: Sequence[float]):
+        """
+        Args:
+            numbers_cliffords: The different numbers of Cliffords in the RB
+                study.
+            ground_state_probabilities: The corresponding average ground state
+                probabilities.
+        """
+        self._num_cfds_seq = numbers_cliffords
+        self._gnd_state_probs = ground_state_probabilities
+
+    @property
+    def data(self) -> Sequence[Tuple[int, float]]:
+        """Returns a sequence of tuple pairs with the first item being a
+        number of Cliffords and the second item being the corresponding average
+        ground state probability.
+        """
+        return [(num, prob) for num, prob in zip(self._num_cfds_seq,
+                                                 self._gnd_state_probs)]
+
+    def plot(self) -> None:
+        """Plots the average ground state probability vs the number of
+        Cliffords in the RB study.
+        """
+        fig = pyplot.figure()
+        pyplot.plot(self._num_cfds_seq, self._gnd_state_probs, 'ro-',
+                    figure=fig)
+        pyplot.xlabel(r"Number of Cliffords")
```

Comment:
Pass the `figure` argument to xlabel and ylabel to avoid touching global state. Also do this in other plot methods.

Reply:
Done.